### PR TITLE
Add tests for funnelshift series math API

### PR DESCRIPTION
**Pull request by 0x12CC**

### CODE REVIEW

File: features/feature_case/math/math-funnelshift.cu [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#include <cassert>
```
Comment (zhimingwang36) [inline features/feature_case/math/math-funnelshift.cu:10]: Please add License header for this new added file.

**Review by ShengchenJ — CHANGES_REQUESTED** [review]
============= math-funnelshift: COMPFAIL ==================
----------------------------
/***/dpct_ci/tst_pub_2/_work/SYCLomatic-test/SYCLomatic-test/c2s_test_repo/test_workspace/features/option_cpu/math-funnelshift/out_root/math-funnelshift.dp.cpp:10:21: error: use of undeclared identifier '__funnelshift_l'
  TestResults[0] = (__funnelshift_l(0xAA000000, 0xBB, 8) == 0xBBAA);
                    ^
/***/dpct_ci/tst_pub_2/_work/SYCLomatic-test/SYCLomatic-test/c2s_test_repo/test_workspace/features/option_cpu/math-funnelshift/out_root/math-funnelshift.dp.cpp:14:21: error: use of undeclared identifier '__funnelshift_lc'
  TestResults[1] = (__funnelshift_lc(0xAA000000, 0xBB, 16) == 0xBBAA00);
                    ^
/***/dpct_ci/tst_pub_2/_work/SYCLomatic-test/SYCLomatic-test/c2s_test_repo/test_workspace/features/option_cpu/math-funnelshift/out_root/math-funnelshift.dp.cpp:18:21: error: use of undeclared identifier '__funnelshift_r'
  TestResults[2] = (__funnelshift_r(0xAA00, 0xBB, 8) == 0xBB0000AA);
                    ^
/***/dpct_ci/tst_pub_2/_work/SYCLomatic-test/SYCLomatic-test/c2s_test_repo/test_workspace/features/option_cpu/math-funnelshift/out_root/math-funnelshift.dp.cpp:22:21: error: use of undeclared identifier '__funnelshift_rc'
  TestResults[3] = (__funnelshift_rc(0xAA0000, 0xBB, 16) == 0xBB00AA);
                    ^
4 errors generated.
----------------------


Case failed.

**Review by ShengchenJ — APPROVED** [review]

File: features/feature_case/math/math-funnelshift.cu [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  constexpr int NumberOfTests = 4;
+  char *TestResults;
+  cudaMallocManaged(&TestResults, NumberOfTests * sizeof(*TestResults));
+  testFunnelShiftKernel<<<1, 1>>>(TestResults);
+  cudaDeviceSynchronize();
+  for (int i = 0; i < NumberOfTests; i++) {
+    if (TestResults[i] == 0) {
+      std::cerr << "funnelshift test " << i << " failed" << std::endl;
+    }
+    assert(TestResults[i] != 0);
```
Comment (ShengchenJ) [inline features/feature_case/math/math-funnelshift.cu]: Please explicit return 0 when the case is pass. And return non-zero when the case failed. 
Currently, if the assert failed. The test driver will exit and the following test suite cases will not execute.

**Review by ShengchenJ — CHANGES_REQUESTED** [review]

File: features/feature_case/math/math-funnelshift.cu (hunk 2 above)
Comment (0x12CC) [reply features/feature_case/math/math-funnelshift.cu]: I've removed the assert and added return statements.

**Review by ShengchenJ — APPROVED** [review]

**Pull request merged**